build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/RestCompatTestTransformTask.java

```diff
@@ -81,14 +89,18 @@
     private final DirectoryProperty sourceDirectory;
     private final DirectoryProperty outputDirectory;
     private final PatternFilterable testPatternSet;
+    private final Factory<PatternSet> patternSetFactory;
     private final List<RestTestTransform<?>> transformations = new ArrayList<>();
+    private final Map<PatternFilterable, String> skippedTestByFilePatternTransformations = new HashMap<>();
+    private final Map<Pair<String, String>, Map<String, String>> skippedTestByTestNameTransformations = new HashMap<>();
```

Comment:
Can we have some comments here so we can understand what this data structure is supposed to represent. We have a map where the key is a `Pair` and the value is yet another map. I had to read the implementation below about 5 times to sort out what's going on here.

Reply:
I think if we implement my suggestion below this will just become a `Map<String, String>` where the key is the test name and the value the reason then we sort everything out later.

Reply:
I implemented something similar to your suggestion, but it still requires a non trival map. However, I did get rid of one of the Pairs and it is more closely aligned with the other file based skip.